pss/keystore.go

```diff
@@ -180,22 +182,22 @@ func (ks *KeyStore) processSym(envelope *envelope) (*receivedMessage, string, Ps
 // encapsulating the decrypted message, and the byte representation of
 // the public key used to decrypt the message.
 // It fails if decryption of message fails, or if the message is corrupted.
-func (p *Pss) processAsym(envelope *envelope) (*receivedMessage, string, PssAddress, error) {
+func (p *Pss) processAsym(pssMsg *message.Message) (crypto.ReceivedMessage, string, PssAddress, error) {
 	metrics.GetOrRegisterCounter("pss.process.asym", nil).Inc(1)
 
-	recvmsg, err := envelope.openAsymmetric(p.privateKey)
+	recvmsg, err := p.Crypto.UnWrapAsymmetric(pssMsg.Payload, p.privateKey)
 	if err != nil {
 		return nil, "", nil, fmt.Errorf("could not decrypt message: %s", err)
 	}
 	// check signature (if signed), strip padding
-	if !recvmsg.validateAndParse() {
+	if !recvmsg.ValidateAndParse() {
 		return nil, "", nil, errors.New("invalid message")
 	}
-	pubkeyid := common.ToHex(p.Crypto.FromECDSAPub(recvmsg.Src))
+	pubkeyid := common.ToHex(p.Crypto.FromECDSAPub(recvmsg.GetSrc()))
```

Comment:
Why do we need a getter, please?

Reply:
Sorry, these changes belong to #1731. This PR shouldn't be reviewed until that one is merged.

Reply:
ReceivedMessage must be an interface because the validation and parsing is implementation-specific.